templates/_base/master/files/etc-kubernetes-manifests-etcd-member.yaml

```diff
@@ -0,0 +1,128 @@
+mode: 0644
+path: "/etc/kubernetes/manifests/etcd-member.yaml"
+contents:
+  inline: |
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: etcd-member
+      namespace: kube-system
+      labels:
+        k8s-app: etcd
+    spec:
+      initContainers:
+      - name: discovery
+        image: "docker.io/abhinavdahiya/origin-setup-etcd-environment" ## FIXME(abhinav): switch this to official image.
+        args:
+        - "--discovery-srv={{.BaseDomain}}"
+        - "--output-file=/run/etcd/environment"
+        - "--v=4"
+        securityContext:
+          priviledged: true
+        volumeMounts:
+        - name: discovery
+          mountPath: /run/etcd/
+      - name: certs
+        image: "quay.io/coreos/kube-client-agent:678cc8e6841e2121ebfdb6e2db568fce290b67d6" ## FIXME(abhinav): these images should be replacable by release image.
+        command:
+        - /bin/sh
+        - -c
+        - |
+          #!/bin/sh
+          set -euo pipefail
+
+          source /run/etcd/environment
+
+          [ -e /etc/ssl/etcd/system:etcd-server:${ETCD_DNS_NAME}.crt -a \
+            -e /etc/ssl/etcd/system:etcd-server:${ETCD_DNS_NAME}.key ] || \
+            /usr/local/bin/kube-client-agent \
+              request \
+                --orgname=system:etcd-servers \
+                --cacrt=/etc/ssl/etcd/root-ca.crt \
+                --assetsdir=/etc/ssl/etcd \
+                --address={{apiServerURL .}} \
+                --dnsnames={{etcdServerCertDNSNames .}} \
+                --commonname=system:etcd-server:${ETCD_DNS_NAME} \
+                --ipaddrs=${ETCD_IPV4_ADDRESS},127.0.0.1 \
+
+          [ -e /etc/ssl/etcd/system:etcd-peer:${ETCD_DNS_NAME}.crt -a \
+            -e /etc/ssl/etcd/system:etcd-peer:${ETCD_DNS_NAME}.key ] || \
+            /usr/local/bin/kube-client-agent \
+              request \
+                --orgname=system:etcd-peers \
+                --cacrt=/etc/ssl/etcd/root-ca.crt \
+                --assetsdir=/etc/ssl/etcd \
+                --address={{apiServerURL .}} \
+                --dnsnames={{etcdPeerCertDNSNames .}} \
+                --commonname=system:etcd-peer:${ETCD_DNS_NAME} \
+                --ipaddrs=${ETCD_IPV4_ADDRESS} \
+          
+        securityContext:
+          priviledged: true
+        volumeMounts:
+        - name: discovery
+          mountPath: /run/etcd/
+        - name: certs
+          mountPath: /etc/ssl/etcd
+      containers:
+      - name: etcd-member
+        image: "quay.io/coreos/etcd:v3.2.14" ## FIXME(abhinav): these images should be replacable by release image.
+        command:
+        - /bin/sh
+        - -c
+        - |
+          #!/bin/sh
+          set -euo pipefail
+
+          source /run/etcd/environment
+
+          /usr/local/bin/etcd \
+            --discovery-srv {{.BaseDomain}} \
+            --initial-advertise-peer-urls=https://${ETCD_IPV4_ADDRESS}:2380 \
+            --cert-file=/etc/ssl/etcd/system:etcd-server:${ETCD_DNS_NAME}.crt \
```

Comment:
Do we ever want `ETCD_DNS_NAME` to differ from the node name? The node name already has to resolve to a routable address.

Reply:
`ETCD_DNS_NAME` this is the name in the SRV record that resolves to this machine's IP.

Reply:
ack